### PR TITLE
Wear configuration demo

### PR DESCRIPTION
**Pull request by danybony**
This PR adds a demo build config needed to add a Wear module to an existing project.
This demo shows how to integrate Wear with some real-world scenario like

- multi-dimensions flavours
- support for alternative markets (without PlayServices)
- different signing keys for different build types

### CODE REVIEW

File: WearBuildConfig/mobile/build.gradle [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        debug {
+            signingConfig signingConfigs.debug
+        }
+
+        release {
+            signingConfig signingConfigs.release
+        }
+
+        qaRelease.initWith(buildTypes.release)
+        qaRelease {
```
Comment (mr-archano) [inline WearBuildConfig/mobile/build.gradle]: maybe just `qa`? I wonder if this will break the smart _task abbreviation_ because of the clashing with `release` 😕

File: LeftNavBarExample/LeftNavBarLibrary/build.gradle [hunk 2]
```diff
@@ -3,7 +3,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:1.0.0'
+        classpath 'com.android.tools.build:gradle:2.2.1'
```
Comment (mr-archano) [inline LeftNavBarExample/LeftNavBarLibrary/build.gradle]: why these changes to other modules? I see we have a root multi-module project, but what's the value of it? ideally all these demos should be separated project, right?
If anything we should consider to extract this common dependency (or at least the version) in a single place where it will be easier to maintain

File: WearBuildConfig/mobile/build.gradle [hunk 3]
```diff
@@ -0,0 +1,71 @@
+apply plugin: 'com.android.application'
+apply from: "$rootProject.projectDir/WearBuildConfig/shared-config/android-signing.gradle"
```
Comment (mr-archano) [inline WearBuildConfig/mobile/build.gradle]: this could be a more idiomatic `rootProject.file('WearBuildConfig/shared-config/android-signing.gradle')`

File: WearBuildConfig/mobile/build.gradle [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    buildToolsVersion "25.0.0"
+
+    defaultConfig {
+        applicationId "com.novoda.demo.wearbuildconfig"
+        minSdkVersion 16
+        targetSdkVersion 25
+        versionCode 1
+        versionName "1.0"
+
+        buildConfigField 'boolean', 'PLAY_SERVICES', "true"
```
Comment (mr-archano) [inline WearBuildConfig/mobile/build.gradle]: if you're seeking a good example of a `build.gradle` you could consider the use of the [`gradle-build-properties-plugin`](https://github.com/novoda/gradle-build-properties-plugin) and its [typed decorators for `buildConfigField`](https://github.com/novoda/gradle-build-properties-plugin#4-typed-buildconfigfield--resvalue)

File: WearBuildConfig/mobile/build.gradle [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+            buildConfigField 'boolean', 'PLAY_SERVICES', 'false'
+        }
+
+        playServices {
+            dimension 'playServices'
+            //uses default PLAY_SERVICES value
+        }
+
+        preProduction {
```
Comment (mr-archano) [inline WearBuildConfig/mobile/build.gradle]: what is `preProduction`? maybe `development` as environment would be easier to grasp

File: WearBuildConfig/mobile/build.gradle [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+configurations {
+    productionPlayServicesReleaseWearApp
+    preProductionPlayServicesReleaseWearApp
+    productionPlayServicesQaReleaseWearApp
+    preProductionPlayServicesQaReleaseWearApp
+}
+
+dependencies {
+    productionPlayServicesReleaseWearApp project(path: ':WearBuildConfigWear', configuration: 'productionRelease')
```
Comment (mr-archano) [inline WearBuildConfig/mobile/build.gradle]: I'm lost. Above you defined 4 additional configurations, but they've not be used anywhere else, right? Maybe this is material for the (missing) `README`

File: settings.gradle [hunk 7]
```diff
@@ -26,3 +26,7 @@ include 'TabHostExample'
 include 'TabHostSelfContainedTabBrowsing'
 include 'wizard'
 include 'EditTextChips'
+include 'WearBuildConfigMobile'
+project(':WearBuildConfigMobile').projectDir = new File('WearBuildConfig/mobile')
```
Comment (mr-archano) [inline settings.gradle]: ok, I think I see what you're trying to do, but I'm still not sure why you need to create this alias (and the one below). If you're trying to workaround the multi module setup in the whole `android-demos` maybe this is not the best solution. clarifying your intent here would help a lot :)

File: LeftNavBarExample/LeftNavBarLibrary/build.gradle (hunk 2 above)
Comment (danybony) [inline LeftNavBarExample/LeftNavBarLibrary/build.gradle]: Regarding the changes, these were done automatically by the IDE, i just thought there was no harm in updating in other modules too.

Regarding the multi-module, I can see it useful to reduce duplication of the gradle wrapper for example, and also the fact that it's easier to load it immediately from the root, having access to all the projects.

This is just my point of view, I don't remember the historical reasons behind this repo structure honestly.

File: WearBuildConfig/mobile/build.gradle (hunk 5 above)
Comment (danybony) [inline WearBuildConfig/mobile/build.gradle]: In this project is to show an example of second flavour dimension: the environment.
Agree, `development` could be easier to understand

File: LeftNavBarExample/LeftNavBarLibrary/build.gradle (hunk 2 above)
Comment (mr-archano) [inline LeftNavBarExample/LeftNavBarLibrary/build.gradle]: My considerations comes from the realisation every module is an app on its own. I see the plus of the simpler import, but that means also longer evaluation times (the whole root project is evaluated in the process 😅 ). I'm not saying you should change the rest of the project, but I would seriously consider not to follow the (broken IMHO) pattern and make this a separate project on its own (all you need is avoid to list your module in the root `settings.gradle`, right?).

Comment (danybony) [inline LeftNavBarExample/LeftNavBarLibrary/build.gradle]: Yes, I see your point.
Honestly I don't mind having this one as an independent project, but that would require a duplication of the gradle wrapper or can we avoid that somehow?

File: WearBuildConfig/mobile/build.gradle (hunk 1 above)
Comment (danybony) [inline WearBuildConfig/mobile/build.gradle]: Good idea for simplify to just `qa`

File: LeftNavBarExample/LeftNavBarLibrary/build.gradle (hunk 2 above)
Comment (mr-archano) [inline LeftNavBarExample/LeftNavBarLibrary/build.gradle]: Nah, duplication of the wrapper can't be avoided, but I really don't feel is like a big deal 😸  Actually this is a good opportunity to use Gradle 3.x. in your own _isolated and cohesive_ project. Mmm 🤔  sounds even better to me.

File: WearBuildConfig/mobile/build.gradle (hunk 6 above)
Comment (danybony) [inline WearBuildConfig/mobile/build.gradle]: These configurations are used under the hood by gradle to copy the Wear APK into the mobile APK's res/raw folder.
This is done ONLY for the configurations here defined, so for example it doesn't happen for `noPlayServices` builds.
Good point on the readme, I'll write one when I'm back

File: WearBuildConfig/mobile/build.gradle (hunk 1 above)
Comment (danybony) [inline WearBuildConfig/mobile/build.gradle]: 6f52498c8dd6198a2f37512127f7425f7ba05397

File: settings.gradle (hunk 7 above)
Comment (danybony) [inline settings.gradle]: What I'm trying to do is to create two modules that share the same parent in the main project root:

- WearBuildConfig/mobile
- WearBuildConfig/wear

Can you specify what's wrong with this way of doing it?
This is the same strategy it was used for other modules here and I'll be curious to hear about alternatives (even if we might make this project independent)

File: WearBuildConfig/mobile/build.gradle (hunk 3 above)
Comment (danybony) [inline WearBuildConfig/mobile/build.gradle]: Done in 3f4cf8d1d9feeb9fbc0481f72c3a28034efae44d

File: settings.gradle (hunk 7 above)
Comment (mr-archano) [inline settings.gradle]: Sorry I wasn't clear enough: read my comment above as something in line with the suggestion of make this module a self-contained project. It's not _wrong_, it follows what's already in place, but I personally think it's a flawed approach that we should address soon.

File: LeftNavBarExample/LeftNavBarLibrary/build.gradle (hunk 2 above)
Comment (danybony) [inline LeftNavBarExample/LeftNavBarLibrary/build.gradle]: Ok, followed your suggestion @mr-archano and moved this demo into a independent project: dec78a29bfe68f851b7a63c06d1dc7fece30a882

File: WearBuildConfig/mobile/build.gradle (hunk 4 above)
Comment (danybony) [inline WearBuildConfig/mobile/build.gradle]: Thanks for the hint, but this field is just an example of a value set differently in the two variations of the flavour dimension.
I'd prefer to keep it as simple as possible, without adding external plugins in this demo

File: LeftNavBarExample/LeftNavBarLibrary/build.gradle (hunk 2 above)
Comment (rock3r) [inline LeftNavBarExample/LeftNavBarLibrary/build.gradle]: Why not updating to 2.2.3 while we're on it? 2.2.1 wasn't the latest even 15 days ago :D

File: WearBuildConfig/gradle/wrapper/gradle-wrapper.properties [hunk 8]
```diff
@@ -0,0 +1,6 @@
+#Tue Dec 06 15:30:18 CET 2016
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-3.0-all.zip
```
Comment (rock3r) [inline WearBuildConfig/gradle/wrapper/gradle-wrapper.properties]: Y U NO [3.2.1](https://docs.gradle.org/current/release-notes)?

File: WearBuildConfig/wear/build.gradle [hunk 9]
```diff
@@ -0,0 +1,38 @@
+apply plugin: 'com.android.application'
+apply from: rootProject.file('shared-config/android-signing.gradle')
+
+android {
+    compileSdkVersion 25
+    buildToolsVersion "25.0.0"
```
Comment (rock3r) [inline WearBuildConfig/wear/build.gradle]: Why not 25.0.1 while we're at it?

File: LeftNavBarExample/LeftNavBarLibrary/build.gradle (hunk 2 above)
Comment (danybony) [inline LeftNavBarExample/LeftNavBarLibrary/build.gradle]: ¯\_(ツ)_/¯ it was not the main point of this PR
